java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -1297,15 +1300,20 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_hash(JNIEnv *env,
 ////////
 
 JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_makeCudfColumnView(
-    JNIEnv *env, jobject j_object, jint j_type, jlong j_data, jlong j_data_size, jlong j_offset,
+    JNIEnv *env, jobject j_object, jint j_type, jint scale, jlong j_data, jlong j_data_size, jlong j_offset,
     jlong j_valid, jint j_null_count, jint size, jlongArray j_children) {
 
   JNI_ARG_CHECK(env, (size != 0), "size is 0", 0);
   try {
     using cudf::column_view;
     cudf::jni::auto_set_device(env);
     cudf::type_id n_type = static_cast<cudf::type_id>(j_type);
-    cudf::data_type n_data_type(n_type);
+    cudf::data_type n_data_type;
+    if ( n_type == cudf::type_id::DECIMAL32 || n_type == cudf::type_id::DECIMAL64 ) {
```

Comment:
We should have a helper function that will take j_type and j_scale and return a data_type, so we can use it everywhere.

Reply:
created helper function which does the above.